library/src/main/java/com/aws/jverify/JVerify.java

```diff
@@ -137,13 +149,15 @@ public static int old(int value) {
     }
 
     /**
-     * TODO decide whether this should exist or not
-     * and if it does add documentation
-     * And even if we remove jequals, it might be good to add documentation on Dafny's == somewhere
-     * To clarify when it resolves to:
-     * - reference equality
-     * - shallow equality
-     * - undefined equality
+     * Provides observational equality. Two values are observationally equal if they can not be distinguished.
+     * For primitive values, jequals behaves the same as `==`
+     * For objects such as records and classes:
+     * - If the type is impure, jequals behaves like `==`, reference equality
+     * - If the type is pure, jequals behaves as structural equality, 
+     * recursively testing for observational equality on all its fields. 
```

Comment:
```suggestion
     *   recursively testing for observational equality on all its fields. 
```
(Actually fairly important to read this line as only applying to "if the type is pure...")